### DISCUSSION TITLE
`wasm-bindgen` version conflict resolution?

### DISCUSSION THREAD

**Discussion by pufferfish101007** [kaleidawave/ezno#222] [Questions]
Hi, I've just discovered ezno and it looks like an absolutely incredible project. I'd quite like to use it in some tests in a project of mine, to verify that some (quite simple) javascript code that I'm producing is type-safe. Unfortunately, my project needs `wasm-bindgen >= 0.2.92`, which conflicts wth `ezno/checker`'s dependency of `wasm-bindgen = 0.2.89`. I don't actually need wasm-bindgen for my tests so I've tried using `cfg` to avoid Cargo trying to load both `wasm-bindgen` and `ezno` at the same time, but it doesn't seem to like it. I'll continue playing around to see if I can get it to work as-is (I'm sure there must be a way), but I would like to ask if there's any way I can help upgrade the `wasm-bindgen` version used in ezno - [checker/Cargo.toml](https://github.com/kaleidawave/ezno/blob/main/checker/Cargo.toml) mentions that it's due to a problem in the `source-map` crate - is there anything I might be able to do to try and help resolve this?
Thanks

**Comment by kaleidawave** [comment]
Hey. Ah yes a while ago I had some problems with getting the `wasm_bindgen_cli` and `wasm_bindgen` crate to have their versions in sync. As a fix I just pinned the version of the crate and CLI. I also applied this pinning fix to the `source-map` crate. 

If I remember correctly, a fix was applied and this pinning was changed to be a *wider version* in #114. As part of it I had to also change the `source-map` crate. It looks like that change was made to the `source-map` crate https://github.com/kaleidawave/source-map/blob/main/Cargo.toml#L23 but I did not go back afterwards and update the requirement in `checker/Cargo.toml`. 

That should be possible to change now and hopefully there are not any breaking changes. Will look at changing the requirement to be `wasm-bindgen = "0.2"` so you don't have any clashes using the crate. Will be part of PR #220.

---

> I'd quite like to use it in some tests in a project of mine, to verify that some (quite simple) javascript code that I'm producing is type-safe

Awesome, let me know how it goes!

**Reply by pufferfish101007** [reply]
That's fantastic, thank you! I found that forking the repo and making the change myself allowed the crate to compile (for the test profile), but I haven't checked to see if it actually works when compiling to wasm (since I don't need to). I haven't actually integrated ezno in my tests yet but I will absolutely let you know how it goes!

**Reply by pufferfish101007** [reply]
> Awesome, let me know how it goes!

I've already had one (very silly) mistake caught by ezno so I'll say it's a success :)

**Reply by kaleidawave** [reply]
Epic! The holidays and now school mean that PR to fix this and the later release will be around 2 weeks now :(.

**Reply by kaleidawave** [reply]
It should be fixed under #220 and in a release in < 1 week (hopefully)
